server/routes/getNewProfile.ts

```diff
@@ -0,0 +1,85 @@
+import { Request, Response, NextFunction } from 'express';
+import { Op } from 'sequelize';
+import { factory, formatFilename } from '../../shared/logging';
+const log = factory.getLogger(formatFilename(__filename));
+import { DB } from '../database';
+
+export const Errors = {
+  NoAddressProvided: 'No address provided in query',
+  NoAddressFound: 'No address found',
+  NoProfileFound: 'No profile found',
+};
+
+const getNewProfile = async (models: DB, req: Request, res: Response, next: NextFunction) => {
+  const { address } = req.query;
+  if (!address) return next(new Error(Errors.NoAddressProvided));
+
+  const addressModel = await models.Address.findOne({
+    where: {
+      address,
+    },
+    include: [ models.OffchainProfile, ],
+  });
+  if (!addressModel) return next(new Error(Errors.NoAddressFound));
+
+  const profile = await models.Profile.findOne({
+    where: {
+      id: addressModel.profile_id,
+    },
+  });
+  if (!profile) return next(new Error(Errors.NoProfileFound));
+
+  const addresses = await models.Address.findAll({
+    where: {
+      profile_id: profile.id,
+    }
+  })
+
+  const chainIds = [... new Set<string>(addresses.map(a => a.chain))]
+  const chains = await models.Chain.findAll({
+    where: {
+      id: {
+        [Op.in]: chainIds,
+      },
+    }
+  })
+
+  const addressIds = [...new Set<number>(addresses.map(a => a.id))]
+  const threads = await models.OffchainThread.findAll({
+    where: {
+      address_id: {
+        [Op.in]: addressIds,  
+      },  
+    },
+    include: [ { model: models.Address, as: 'Address' } ],
+  });
+
+  const comments = await models.OffchainComment.findAll({
```

Comment:
This is fine for now but in the future we may want to consider paginating these calls, if loading time becomes a problem for very active users (or bots).